core/src/test/java/org/ehcache/EhcacheBasicPutIfAbsentEventsTest.java

```diff
@@ -0,0 +1,164 @@
+    ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+  }
+
+  @Test
+  public void testPutIfAbsentKeyNull() {
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    try {
+      ehcache.putIfAbsent("key", null);
+      fail();
+    } catch (NullPointerException e) {
+      // expected
+    }
+    verify(cacheEventListener,never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+  }
+
+  @Test
+  public void testPutIfAbsentNullValue() {
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+
+    try {
+      ehcache.putIfAbsent(null, "value");
+      fail();
+    } catch (NullPointerException e) {
+      // expected
+    }
+    verify(cacheEventListener,never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+  }
+
+  @Test
+  public void testPutIfAbsent() throws Exception {
+    final FakeStore fakeStore = new FakeStore(Collections.singletonMap("key", "oldValue"));
+    this.store = spy(fakeStore);
+    doThrow(new CacheAccessException("")).when(this.store).computeIfAbsent(eq("key"), getAnyFunction());
+    final Ehcache<String, String> ehcache = this.getEhcache(this.cacheLoaderWriter);
+    ehcache.putIfAbsent("key", "value");
+    verify(cacheEventListener,times(1)).onEvent(argThat(isCreated));
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e){
+      //expected
+    }
+  }
+
+  @Test
+  public void testPutIfAbsentNoStoreEntryCacheAccessExceptionNoCacheLoaderWriter() throws Exception {
+    final FakeStore fakeStore = new FakeStore(Collections.<String, String>emptyMap());
+    this.store = spy(fakeStore);
+    doThrow(new CacheAccessException("")).when(this.store).computeIfAbsent(eq("key"), getAnyFunction());
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    ehcache.putIfAbsent("key", "value");
+    verify(cacheEventListener,never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e){
+      //expected
+    }
+  }
+
+  @Test
+  public void testPutIfAbsentNoStoreEntryCacheAccessExceptionHasCacheLoaderWriterEntry() throws Exception {
+    final FakeStore fakeStore = new FakeStore(Collections.<String, String>emptyMap());
+    this.store = spy(fakeStore);
+    doThrow(new CacheAccessException("")).when(this.store).computeIfAbsent(eq("key"), getAnyFunction());
+    final Ehcache<String, String> ehcache = this.getEhcache(this.cacheLoaderWriter);
+    ehcache.putIfAbsent("key", "value");
+    verify(cacheEventListener,times(1)).onEvent(argThat(isCreated));
```

Comment:
the missing space makes me think that the formatting of the whole PR should be double-checked